Source/WebCore/Modules/cookie-store/CookieStore.h

```diff
@@ -62,6 +63,10 @@ class CookieStore final : public RefCounted<CookieStore>, public EventTarget, pu
 private:
     explicit CookieStore(Document*);
 
+    // CookieChangeListener
+    void cookiesAdded(const String& host, const Vector<Cookie>&) final;
+    void cookiesDeleted(const String& host, const Vector<Cookie>&) final;
+
     // ActiveDOMObject
     const char* activeDOMObjectName() const final;
```

Comment:
You'll need to override ActiveDOMObject::virtualHasPendingActivity() and return true when `m_hasChangeEventListener` is set, to keep the JS wrapper alive when there is an active cookie change listener.

Reply:
Oh I see. I'm just curious. The JS wrapper for what? And also, where does `ActiveDOMObject::virtualHasPendingActivity()` get called?

Reply:
This object is exposed to Javascript so whenever the JS interacts with this object, they interacts with its JS wrapper and the wrapper basically converts JS calls into C++ calls.

It is a requirement for the JS wrapper to exist to be able to dispatch events on it. hasPendingActivity() will prevent garbageCollection of the JS wrapper as long as we return true.